core/src/main/scala/com/nvidia/spark/rapids/tool/tuning/AutoTuner.scala

```diff
@@ -850,6 +896,27 @@ abstract class AutoTuner(
             }
             appendRecommendationForMemoryMB("spark.executor.memory",
               s"${recomMemorySettings.executorHeap.get}")
+
+            // Add off-heap memory recommendation based on hybrid scan detection
+            val offHeapSizeMB = platform.getUserEnforcedSparkProperty("spark.memory.offHeap.size")
+              .map(StringUtils.convertToMB(_, Some(ByteUnit.BYTE)))
+              .getOrElse(calculateOffHeapMemorySize(execCores))
+            if (offHeapSizeMB > 0) {
+              appendRecommendationForMemoryMB("spark.memory.offHeap.size", s"$offHeapSizeMB")
+              // Enable off-heap memory if we're recommending a size
+              appendRecommendation("spark.memory.offHeap.enabled", "true")
+
+              // Calculate host off-heap limit size for onPrem platform only when
+              // offHeapLimit is enabled
+              if (!platform.isPlatformCSP && isOffHeapLimitEnabled) {
+                val hostOffHeapLimitSizeMB = recomMemorySettings.executorMemOverhead.get +
```

Comment:
In line 907 we are recommending `spark.memory.offHeap.enabled` to `true` and then in line 911 we are checking `isOffHeapLimitEnabled`.

Can you check if one of it is redundant?